### PR TITLE
Display an error when trying to add an user with a non-localhost IP

### PR DESCRIPTION
**Pull request by ghost**
## Purpose

When trying to call `add_api_user` from a non-localhost IP, no error message is sent to the client, nor logged. This emits an error message using the custom error handling from #847.

## Context

Emit an error when calling `add_api_user` from a remote IP.

## Changes

Update the `add_api_user` method and update the API documentation with an output example.

## How to test this PR

Calling the `add_api_user` from a remote IP should send back an error message. For instance `curl -s "Content-Type: application/json" --data '{"method":"add_api_user","params:{"username":"toto","api_key":"toto"},"id":1,"jsonrpc":"2.0"}' http://10.0.0.1:5000/  | jq` should return something like
```
{
  "id": 1,
  "error": {
    "message": "Unauthorized to call this method from a remote IP",
    "data": {
      "remote_ip": "10.0.0.2"
    },
    "code": -32603
  },
  "jsonrpc": "2.0"
}
```

### CODE REVIEW

File: lib/Zonemaster/Backend/RPCAPI.pm [hunk 1]
```diff
@@ -575,6 +575,12 @@ sub add_api_user {
         if ( $allow ) {
             $result = 1 if ( $self->{db}->add_api_user( $params->{username}, $params->{api_key} ) eq '1' );
         }
+        else {
+            die Zonemaster::Backend::Error::PermissionDenied->new(
+                message => 'Unauthorized to call this method from a remote IP',
```
Comment (matsduf) [inline lib/Zonemaster/Backend/RPCAPI.pm:580]: I think that "forbidden" or "not permitted" is better than "unauthorized" in this context. There is no authorization mechanism for this. Suggestions:
```
'Call to this method from a remote IP is not permitted'
```

File: docs/API.md [hunk 2]
```diff
@@ -1208,9 +1208,15 @@ Omitting params:
...
-  "result": 0,
   "id": 1,
-  "jsonrpc": "2.0"
+  "jsonrpc": "2.0",
+  "error": {
+    "code": -32603,
+    "data": {
+      "remote_ip": "10.0.0.1"
+    },
+    "message": "Unauthorized to call this method from a remote IP"
```
Comment (matsduf) [inline docs/API.md:1218]: See comment in RPCAPI.pm.

File: lib/Zonemaster/Backend/RPCAPI.pm (hunk 1 above)
Comment (pnax) [reply lib/Zonemaster/Backend/RPCAPI.pm]: Changed to:
```
Call to "add_api_user" method not permitted from a remote IP
```

**Comment by matsduf** [comment]
## Testing v2021.2
LGTM.

**Pull request merged**